lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -426,3 +434,21 @@ def _extend_sequential_info_with_assignment_progress(
                         }
                     }
                 )
+
+    @staticmethod
+    def _extend_block_info_with_offline_data(blocks_info_data: Dict[str, Dict]) -> None:
+        """
+        Extends block info with offline download data.
+
+        If offline content is available for the block, adds the offline download data to the block info.
+        """
+        for block_id, block_info in blocks_info_data.items():
```

Comment:
Can we optimize it by filtering only for supported offline types? 
TBD in follow up